airflow/providers/databricks/sensors/databricks_partition.py

```diff
@@ -0,0 +1,167 @@
+        return sql_result
+
+    def _generate_partition_query(
+        self,
+        prefix: str,
+        suffix: str,
+        joiner_val: str,
+        table_name: str,
+        opts: dict[str, str] | None = None,
+        escape_key: bool = False,
+    ) -> str:
+        partition_columns = self._sql_sensor(f"DESCRIBE DETAIL {table_name}")[0][7]
+        self.log.info("table_info: %s", partition_columns)
+        if len(partition_columns) < 1:
+            raise AirflowException("Table %s does not have partitions", table_name)
+        formatted_opts = ""
+        if opts is not None and len(opts) > 0:
+            output_list = []
+            for partition_col, partition_value in self.partitions.items():
+                if escape_key:
+                    partition_col = self.escaper.escape_item(partition_col)
+                if partition_col in partition_columns:
+                    if isinstance(partition_value, list):
+                        output_list.append(f"""{partition_col} in {tuple(partition_value)}""")
+                    if isinstance(partition_value, (int, float, complex)):
+                        output_list.append(
+                            f"""{partition_col}{self.partition_operator}{self.escaper.escape_item(partition_value)}"""
+                        )
+                    if isinstance(partition_value, (str, datetime.date)):
+                        output_list.append(
+                            f"""{partition_col}{self.partition_operator}{self.escaper.escape_item(partition_value)}"""
+                        )
+                    # TODO: Check date types.
+                else:
+                    raise AirflowException(
+                        "Column %s not part of table partitions: %s", partition_col, partition_columns
+                    )
+        self.log.debug("Formatted options: %s", formatted_opts)
+        formatted_opts = f"{prefix} {joiner_val.join(output_list)} {suffix}"
+
+        return formatted_opts.strip()
+
+    def _check_table_partitions(self) -> list:
+        _fully_qualified_table_name = str(self._catalog + "." + self._schema + "." + self.table_name)
+        self.log.debug("Table name generated from arguments: %s", _fully_qualified_table_name)
+        _joiner_val = " AND "
+        _prefix = f"SELECT 1 FROM {_fully_qualified_table_name} WHERE"
+        _suffix = " LIMIT 1"
+
+        partition_sql = self._generate_partition_query(
+            prefix=_prefix,
+            suffix=_suffix,
+            joiner_val=_joiner_val,
+            opts=self.partitions,
+            table_name=_fully_qualified_table_name,
+            escape_key=False,
+        )
+        return self._sql_sensor(partition_sql)
+
+    def _get_results(self, context: Context) -> bool:
+        result = self._check_table_partitions()
+        self.log.debug("Partition sensor result: %s", result)
+        if len(result) < 1:
+            raise AirflowException("Databricks SQL partition sensor failed.")
```

Comment:
this handles lack of results differently than the generic sensor.

Reply:
Unified the handling across classes.